### PR TITLE
remove browser support

### PR DESCRIPTION
**Pull request by rochdev**
### What does this PR do?

Remove browser support.

### Motivation

Browser support was experimental and is now deprecated since this is now handled by the [browser SDK](https://github.com/DataDog/browser-sdk) for [RUM](https://docs.datadoghq.com/real_user_monitoring/). Removing this will also greatly simplify internal module load order which will make it easier to implement dynamic configuration and implicit initialization of the tracer.

### CODE REVIEW

File: packages/dd-trace/test/instrumenter.spec.js [hunk 1]
```diff
@@ -79,6 +79,8 @@ describe('Instrumenter', () => {
       .forEach(name => {
         delete require.cache[name]
       })
+
+    process.env.DD_TRACE_DISABLED_PLUGINS = 'mocha'
```
Comment (bengl) [inline packages/dd-trace/test/instrumenter.spec.js:83]: I think you meant to delete mocha from this?

File: packages/dd-trace/src/tracer.js [hunk 2]
```diff
@@ -16,8 +16,10 @@ class DatadogTracer extends Tracer {
   constructor (config) {
     super(config)
 
+    const Scope = getScope()
```
Comment (bengl) [inline packages/dd-trace/src/tracer.js:19]: Doesn't a `name` need to be passed in here?

**Review by bengl — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/tracer.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ class DatadogTracer extends Tracer {
   constructor (config) {
     super(config)
 
-    const Scope = getScope()
+    const Scope = getScope(config.scope)
```
Comment (bengl) [inline packages/dd-trace/src/tracer.js:19]: Is there a test covering this? It seems like this should make or break a test

Comment (rochdev) [reply packages/dd-trace/src/tracer.js:19]: Actually, this should have made the integration tests fail, not sure why this wasn't the case, I'll look into it.

Comment (rochdev) [reply packages/dd-trace/src/tracer.js:19]: Ah, it's because they are not merged yet: https://github.com/DataDog/dd-trace-js/pull/1224

Comment (bengl) [reply packages/dd-trace/src/tracer.js:19]: Ah yeah, in that case, disregard.

**Review by bengl — APPROVED** [review]

**Pull request merged**